ui/src/actionBar.tsx

```diff
@@ -1,19 +1,24 @@
 import ArrowBackIcon from "@mui/icons-material/ArrowBack";
-import FileDownloadIcon from "@mui/icons-material/FileDownload";
+import FileDownloadIcon from "@mui/icons-material/Download";
 import AppBar from "@mui/material/AppBar";
 import Box from "@mui/material/Box";
 import IconButton from "@mui/material/IconButton";
 import Toolbar from "@mui/material/Toolbar";
 import Typography from "@mui/material/Typography";
 import * as React from "react";
 import { Link as RouterLink } from "react-router-dom";
+import { Dataset } from "./dataset";
+import { useSqlDialog } from "./sqlDialog";
 
 type ActionBarProps = {
   title: string;
   backUrl?: string;
+  dataset: Dataset;
 };
 
 export default function ActionBar(props: ActionBarProps) {
+  const [dialog, show] = useSqlDialog({ dataset: props.dataset });
```

Comment:
nit: maybe `showDialog`? In the scope of the action bar component, it's not clear what a variable named `show` might be showing